---
title: "Job Seq: The Lowest Throughput Sequencing Technology"
date: 2018-04-09T12:08:50-07:00
draft: true
tags: [job, phd, bioinformatics, computational biology]
categories: []
---

[image: ]

This post is about getting my experience in a job after finishing my PhD. I've been asked multiple times by several people so it's [time to write a blog post](https://twitter.com/drob/status/928447584712253440).

In fact, I gave a [talk](https://twitter.com/olgabot/status/969649021345083398) about it at UC Davis (slides below). If you're interested, [here is the video](https://www.youtube.com/watch?v=GzbWAh4D4C8).

<iframe src="https://www.slideshare.net/olgabotvinnik/slideshelf" width="615px" height="470px" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" style="border:none;" allowfullscreen webkitallowfullscreen mozallowfullscreen></iframe>

Before I get started I should explain what [open science](https://en.wikipedia.org/wiki/Open_science) means to me. I believe that science will advance if everyone is given access to the data and code behind the result as soon as possible. I don't want to wait for the paper to be published to use your code or play with your data. I want to see it *before* it's publication-ready. I don't want to spend my time reinventing the wheel, rewriting code you already wrote just because you didn't share it. Thus, joining a place that had open science as a core value was very important to me.

Spoiler alert: I eventually ended up at [Chan Zuckerberg Biohub](https://www.czbiohub.org/).


## Rough Timeline

Here's a rough timeline of my job search:

- *December 2017* - [Prepared CV](https://github.com/olgabot/latex-moderncv)
- *January 2018* - Applied to postdocs, then scrapped that and applied to industry jobs
- *February 2018* - More industry aplications
- *March 2018* - Phone interviews
- *April 2018* - Onsite interviews
- *May 2018* - [Defended](https://twitter.com/olgabot/status/863583986970353664?lang=en) May 12th & started new job May 29th!

## Prepping CV+Cover Letter and applying to postdocs

Over the holiday break, I went home to family in Oregon and started preparing my CV and writing cover letters. In preparing my CV, I felt super inadequate. I felt my CV didn't represent all the work I did over the past 4 years - I didn't have enough publications, I didn't have a Nature/Science/Cell paper, the usual academic woes. The worst was when I started comparing myself to people from college and found an outlier who was already on the *tenure-track job market*, so basically light-years ahead of where I was. I felt like that could have been me if things had worked out differently for me.

I decided if I was going to do a postdoc, it had to follow the following parameters, in priority order (most imoprtant first):

1. Be in a location near a Google office that had my partner's product area. This was extremely high priority for me as we had spent the entirety of my post-undergrad career as a long-distance relationship.
1. Be in a lab or environment that was pro-open science, pro-open source software development. This means posting preprints, openly developing code on GitHub before the paper is published, and posting the data before the paper is published.
1. Work on single-cell analyses, since I wanted to continue studying molecular and cellular heterogeneity.
1. PI was an HHMI Investigator (proxy for quality) **and** was vetted as being a great person to work with, and hired wonderful people. Basically, I created a shortlist of people I wanted to work with and emailed it out to previous mentors who pointed out who was both a "great person and great scientist."
1. Ideally, the lab would write "simple" methods papers that actually explain the math behind the paper, rather than shoving all the equations to the supplementary text which is a pet peeve of mine. ðŸ˜ž

Given these constraints, the only way I felt the open science and open source criteria could be reliably enforced was to be part of either the [University of Washington (UW) eScience Institute](http://escience.washington.edu/get-involved/postdoctoral-fellowships) or the [Berkeley Institute for Data Science (BIDS)](https://bids.berkeley.edu/call-data-science-fellow-applications) Postdoctoral Fellowship programs. I also considered applying to the [HHMI Hanna H. Gray Fellows Program](http://www.hhmi.org/programs/hanna-h-gray-fellows-program) since it would allow me to pursue my open science agenda regardless of the advisor. But when I talked to potential advisors about it, turns out they there's a limit of one person per lab and they already had someone applying.

*Side note:* Advisors expect your cover letter + CV as a single document. My [LaTeX ModernCV GitHub Repo](https://github.com/olgabot/latex-moderncv) does this, and [here](https://github.com/olgabot/latex-moderncv/blob/master/potential-advisors/example-advisor.tex) is an example cover letter. It follows the ["Salutation, me, you, us together, talk to you soon, closing" format](https://academia.stackexchange.com/a/5513).

Since the eScience application deadline was in January, I emailed potential mentors at UW and started writing an application right away. It some time to hone the proposal and I was quite proud of the final product. The BIDS deadline wasn't for a few months but I sought out mentors at UCSF/Berkeley, and one of them mentioned that Chan Zuckerberg (CZ) Biohub may be a better fit for me.

Three important pieces of information surfaced during my postdoc search:

1. Chan Zuckerberg Initiative (CZI), announced a few months before I started applying, had claimed they would be pro-open science and open source software development. I didn't believe that they were serious about open science until I heard through my network that they hired [Jeremy Freeman](http://www.jeremyfreeman.net/), a friend of mine from the open science/open source world to head their computational biology. Then I believed that they were serious about making a difference in scientific research through open source and open science.
1. CZ Biohub was going to be doing both the in-the-trenches biological research *and* be pushing the boundaries of open science.
1. I could get paid an industry PhD-level salary rather than a postdoc one, without compromising my values oof open science and open source software.

So I stopped applying to postdocs.

## No more postdocs

At the industry job, I still wanted to do scientific research. Keeping this in mind, here was my criteria for an industry job:

1. Pro-open source, pro-open science. The company needed to publish its code as open-source software so others may inspect and improve upon it.
1. Small, Startup-ish, <200 people. I wanted to be in a place that I could be a part of the founding team and influence the culture.
1. Communicate scientific findings outside of the company. Meaning, the science doesn't stay locked within the company. This could be papers or blog posts.


Of the ~10 companies I applied to, I actually heard back from 7. The companies for which I have nothing negative to say, are named, whereas the rest are anonymized.

1. Startup A
1. Startup B
1. Startup C
1. [Synthego](http://www.synthego.com/)
1. [Immuneering](http://immuneering.com/)
1. [Chan Zuckerberg Initiative](https://www.chanzuckerberg.com/)
1. [Chan Zuckerberg Biohub](https://www.czbiohub.org)

At all the places I applied, the job title was approximately "Bioinformatics Scientist," "Computational Biologist," or "Biological Data Scientist."

### Overall job application + interview flow

1. Phone interview with recruiter
  - Send thank you email within 24 hours!
1. Technical phone interview with hiring manager
  - Send thank you email within 24 hours!
1. Onsite interview, typically with a talk
  - Send thank you email within 24 hours!


Below are some example "Thank you" emails I sent for my Biohub stuff.

#### Example "Thank you" email for phone interview

> Hi Jim, 
> Thank you do much for taking with me yesterday! I'm very excited about Biohub's work and hope I can be a part of it.
> Warmest,
> Olga

#### Example "Thank you" email for onsite interview

> Hi Jim, 
> Thank you again for meeting with me today. You and everyone else have confirmed that CZ Biohub is the incredible, unique, open, and ambitious place I thought it was. I'm so excited to hear back from you. 
> Warmest, 
> Olga

### Startup A

I didn't get past the technical interview here. The interviewer asked me about alignment and seemed unconvinced that I knew anything, and asked me to explain how Smith-Waterman local alignment works. I said "okay..." and dug into my memory from four years ago (first year of grad school)! 

They seemed annoyed and unimpressed with my explanation and demanded, 

> (Interviewer): What is the difference between a dynamic programming algorithm and a regular one?
>
> (Me): You're constantly updating the latest answer.
>
> (Interviewer): No. The Nth answer depends on the (N-1)th answer.

In my mind, I loudly sighed and said "That's basically what I said, but not in a formula!!" Anyway, there were a bunch of red flags in the interview:

- Interviewer shows off that they're smarter than you
- When I asked, "What's your management style?" they responded, "What do you mean?" And suddenly I had to explain what their job really was... The eventual answer was still bad, "Well I stand around at my desk and people ask me questions." No proactive interaction with subordinates, no plan for team building and within-team interaction, nothing.

So really, I was glad they didn't think I was a good "fit."

### Startup B

This place touted itself as attracting amazing scientists, and certainly both the academic pedigree (where they trained) and the scholarly pedigree (where they published) were impeccable by modern standards. But science is not enough.

My talk went okay. They seemed unimpressed by my computational methods and I had to pull out my supplementary slides to show the rigorous testing I did.

Meeting with the C-level (CEO, Chief Scientific Officer, etc) people, they touted the company as ["Bell Labs"](https://en.wikipedia.org/wiki/Bell_Labs) for biology, meaning a place that would lay the groundwork for the next century of biological research. They said there's no place like here, that I would have freedom to work on cutting edge scientific problems with incredible compute resources. One of the C-levels had orchestral music playing in their office and I said it sounded nice. Since in my "about me" slide, I mention that I play cello, they asked if I knew the piece and was shocked and disappointed that I didn't know this opera or violin concerto - whatever it was, it wasn't a cello piece that I would have listened to so I can play it better. Anyway, they made me feel bad and stupid that I didn't know it was XYZ concerto and that I didn't prolifically listen to classical music outside of my repertoire, but look... I am also a hip hop dancer. I listen to hip hop and R & B, and I don't have time to listen to every classical piece out there! I played it off as I wasn't as "cultured" as them.

From the my-level people, I got a different story. Everyone who was there for three or fewer months was really happy and excited about the potential research. There was only one person who has been there longer, and they took me outside for a walk (red flag: didn't want to be near coworkers), and gave me the real scoop. They told me that they're basically writing single-cell RNA-seq pipelines (aka really boring work) instead of exploring scientific questions, and that the leadership doesn't really listen to them.


I did not get an offer from this company.

Red flags:

- Nearly everyone was from the same two R1 schools as the C-levels, suggesting to me that one or both of these were true, and neither of which were any good:
  1. They hired their friends
  2. They were elitist and didn't see people outside of their circle as being valuable
- Interviewer tries to be smarter than you (the concerto incident)
- C-level people are happy but after a 3-month "honeymoon" period, my-level people weren't

### Startup C

The phone interview went well, though they asked some odd questions like "Name some Python packages you use," presumably to filter out people who put *Python - Expert* on their CV but don't use it often or know how to write a Python package.

My talk went well, they really liked it. We then had lunch at a nearby restaurant, where it was kinda awkward. It seemed like they didn't know each other that well so I asked if they go out to eat together often and got mumbles in response. It seemed they only went to eat as a team to put on a show for interviews but didn't normally hang out together. It may seem minor to some, but to me it's important to be polite to waitstaff and say "Thank you" when they refill your water or serve your food. I noticed I was the only one saying "thank you" when they refilled my water, and I was the first to thank the restaurant as we were leaving, then the rest of the group chimed in.

During the interview, I also got some simple statisical questions about how many times you'd need to flip a coin to know if it's biased or not, which is something you learn very early on in grad school. I pointed out that if it's not very biased (e.g. 55:45), you'd have to do a lot more trials than if it was 99:1, which the interviewer said most people don't pick up on (??)

The manager was great. They had a really fleshed out plan for how they wanted the team to grow, and were proactive in making sure their people were taken care of. 

While on the people side, the pros outweighed the cons, this company wasn't huge on open source. They had some code "available" but it was just a source download, there were a bunch of mystery binaries to create their proprietary formats, and there was no way for users to collaboratively submit changes upstream via GitHub or BitBucket. They claimed it was a "support burden."

I ended up getting an offer from this company.

#### Red flags:

- Team didn't seem to know each other
- Not as polite as I wanted to waitstaff
- Weirdly simple questions on interview
- Claimed to be open source but wasn't really

### [Synthego](http://www.synthego.com/)

Synthego makes synthetic oligonucleotides ("oligos"), primarily for CRISPR/Cas9 experiments. By purchasing oligos from Synthego, scientists typically save months of time that they'd otherwise spend biologically synthesizing (and debugging...) the oligos themselves through bacteria and cell lines.

The Chief Scientific Officer (CSO), [Rich Stoner](http://richstoner.github.io/#/) and I had met at a [Code Neuro](http://codeneuro.org/) conference. I contacted him when I was job hunting and applied to their "Exceptional Human" job posting.

They were in the process of growing from 20-100 people, which apparently is the [most difficult thing for a startup to do](https://getlighthouse.com/blog/company-growth-everything-breaks-25-employees/) (see also: http://firstround.com/review/The-Dos-and-Donts-of-Rapid-Scaling-for-Startups/) I got a real taste of their growth craziness because they were in the middle of moving from one building to another. People were literally sitting at completely different desks from when my interview started and ended!

Talking with the C-level people, they really knew what they were doing. It's not their first startup. They came from both tech (SpaceX) and biotech (Zymergen, Applied Biosystems) startups and knew what it took to grow a company. They had exceptional answers to "how will you maintain culture when you grow?" and other growth questions. They had amazing vision for where they saw the company being in 5, 10 years and I truly admired them for their gusto.

Talking with their CFO was confusing for me because they said Silicon Valley things like "this is a high-growth opportunity," aka you can make lots of $$ if you get in now. Since I really didn't know what to ask, I asked, "What questions would you ask if you were in my position?" which worked well.

Since I had a lot of personal goals I wanted to achieve, specifically with dance -- I tried out for the Golden State Warriors Dance Team in 2017 and am currently on a [hip hop dance company](https://www.facebook.com/SoulForceDanceCo/) -- and I wasn't ready to sacrifice them for the love of the startup. So we mutually agreed not to pursue an offer.

#### Green flags:

- Great people, great team
- Not the founders' first startup
- Liked the answers they gave for "how will you maintain culture when you grow?"
- Awesome projects and vision for the future


### [Immuneering](http://immuneering.com/)
Immuneering is a small bioinformatics consulting company in Boston and New York that works with academics and industry. Recently, they have a [paper in neurodegenerative disease](https://www.nature.com/articles/srep14324) that I found extremely interesting because it used openly available data and highlighted a gene I knew and loved: *SNAP25*.

One of the VPs, [Rebecca Kusko](https://www.linkedin.com/in/beckykusko) and I were homework buddies back in undergrad. She finished her PhD a few years before me and sent me an annual job posting request which I sent out to our grad student mailing list. Also, I had reached out to the founder before I left Boston and expressed interest in working for Immuneering, but I was on my way to UC Santa Cruz so it didn't make sense.

The phone and onsite interview went great. They flew me out to Boston for the onsite and I really loved the team. During my talk, I made sure to point out one of the "interesting" genes in our research, *SNAP25* that they also recently published on. I saw the room crack a smile when that slide went up :)

The team dynamic was really great. They truly knew each other and cracked small jokes together. They described the end of a project as "appreciation" ping pong which went something like this:

> Someone: Sansa, that plot really drove the point home to illuminate the biological result
>
> Sansa: Oh, I couldn't have done it without Arya's analysis of the literature
> 
> Arya: Oh, I couldn't have done it without Ygritte's plotting package
>
> Ygritte: Oh, I couldn't have done it without Danerys's package manager

... you get the point. I loved that they were so vocal and appreciative of each others' contributions and that it wasn't a solo effort.

Since location was extremely important to me and they currently had positions in Boston and New York, but not San Francisco, we mutually decided not to pursue an offer.

#### Green flags:

- "Appreciation ping-pong"
- Published papers on their findings using openly available data
- Women in prominent positions in the organization

### [Chan Zuckerberg Initiative](http://chanzuckerberg.com) (CZI)

CZI was actually the first place I applied. 

A few months before I started applying to postdocs, I watched the [CZI announcement]() in October 2016 -- plus anything else I could get my hands on about the organization, e.g. this town hall by Joe Derisi about the CZ Biohub Investigators -- and was compelled by the lofty mission of curing all disease by 2100. It seems an outrageous goal to most, but as Cori Bargmann put the last 100 years of medical advancses into perspective, I believed it was possible. But I knew that in the current climate of "Publish or Perish," and the [overly strong] reliance on publishing in high ["impact factor"](link to titus brown post on how impact factors are bogus) journals for promotion and grants was BS and the system needed a jump start. The fact that [CZI donated to the biology preprint server BioRxiv](link) was an indication to me that they actually "walked" the talk of open science, but I wasn't convinced from an open source perspective. I hadn't heard of Cori Bargmann before and I didn't know her as publishing open source software for scientific researchers.

Fast forward to January 2017, and I participated in the, stay with me here, ["Reproducible Research Curriculum Hackathon"]() led by [Tracy Teal]() and [Erin Becker]() of Data Carpentry. At that event, I found out through the open source grapevine that Jeremy Freeman had been hired at CZI as their Directory of Computational Biology. I knew Jeremy through [Nick Sofroniew](), with whom I did a [summer research program at Janelia Farm](). Nick invited me to speak at the [CodeNeuro]() conference he and Jeremy were organizing in New York City. I agreed, and then liked the conference so much that I [blogged about it]() and helped with organizing the San Francisco version of the conference. I liked the structure -- there were enough talks but not too many, there was enough time for socializing, and importantly there was time for actual *coding* on the second day of the conference. I taught a [gitgoing]() tutorial the first year with [Ben Sussman]() and an [Advanced Pandas]() tutorial the second year. So I knew that Jeremy was dedicated to open source software in science, and started googling "Jeremy freeman chan zuckerberg initiative" like crazy, and applied to the "Computational Biologist" job posting on CZI before I realized I could just email him. I heard back from him and got started with the hiring process.

In Februrary I had my phone interview with the recruiter. He was a real "recruiter" in that he was way chattier than anyone I had encountered in the past few years I was in grad school. He pointed out that I volunteered with "Science Club for Girls" in Boston where I co-taught a biology afterschool curriculum to 2nd graders, which aligns with the overall mission of CZI. I liked that he said they have a "no asshole rule,"  -- when I told a to-be-unnamed family friend about this "rule", said that his job should at least have a "no *more* assholes rule" -- and it seemed they are really focused on having great people who are aligned with the mission, rather letting people through if they are "rockstars," even if they have bad people skills.

For the onsite interview, I prepared for a technical software engineer interview with my partner, who is a software engineer at Google. I worked through exercises in [Cracking the Coding Interview by Laakman-McDowell](), and refreshed my memory of powers of 2. Since I had the interview schedule and names of my interviewers, I googled them and took some notes so I knew who would be giving me technical questions. We role-played - my partner pretended to be one of my interviewers, and I wrote out the code on a whiteboard. It helped a lot to boost my confidence for the interview, because I otherwise felt totally unprepared for technical coding quesions.

The actual onsite interview came a few weeks later. The morning of, my interview started at 11am but I woke up at 7. I did a workout on YouTube and had a leisurely breakfast at a cafe nearby where I met a nice woman from Alaska and told her this was my dream job. I was done by 10 and was so nervous before my interview that I needed to walk off my nervous energy. I walked up and down University Ave in Palo Alto three times until 10:30, when I felt it was reasonable to come in to my interview. 

During the interview, I didn't have any technical coding questions! I talked about my grad school research and my passion for open source software, the Python packages I had written, and what I thought the [Human Cell Atlas](http://humancellatlas.org) (a project that CZI invested in) would achieve for health. It seemed like a really good fit culture and mission-wise, but my interviewers said things I didn't fully understand about the job, such as that I would "define strategy" which sounded weird and managementy to me. I slowly realized that the job they wanted me to do more high-level than I was used to, and that I wouldn't be doing so much actual research in my day-to-day.

I followed up with a thank you email to my recruiter, who (I hope) passed it along to my interviewers. I wasn't sure how to go from there because I really wanted to continue doing research. I ended up applying to the CZ Biohub Computational Biologist job posting in the interim. Eventually, I got lunch with Jeremy and we talked more about the job and I said at this point, I'm more interested in doing the research itself rather than seeing it from afar. 

Since I wanted to be closer to the research and actually get my hands dirty with data and code, we mutually decided that the Biohub would be a better fit for me, and so I didn't get an offer.

#### Green flags:

- "No asshole rule"
- President of Science is a woman (arbitrary to some but important to me)

### [Chan Zuckerberg Biohub](http://czbiohub.org/) ("Biohub")

Honestly, this was my "dream" and "reach" job that I totally didn't think I was going to get.

I applied to the "Computational Biology" position with a combined [Cover Letter + CV](https://github.com/olgabot/latex-moderncv) pdf (like I did to everything else). Jim Karkanias (now my boss!), the head of Data Sciences, got back to me and we scheduled a "phone" interview over Skype. I emphasized my interest in open source software, open science, and working on Cell Atlas-style projects. Again, it was a really good fit culture and mission-wise so I was really excited to hear about the potential projects. The team was also just Jim at the time, so I was excited to get on the ground floor of the organization and push my own open science agenda. I followed up with a thank-you email that day.

I was excited to receive an invitation for an onsite interview. The morning of the interview, I scheduled [1.5 hour without interruptions](https://github.com/tjluoma/John-Cleese-on-Creativity/blob/master/Transcript.markdown#set-aside-about-90-minutes-not-all-morning-1653) (no messages of any kind - no email, no phone) to work on my dissertation at the nearby UCSF Children's Hospital. Once it finished, I checked my email and saw that one of the people I was interviewing with, Steve Quake, had to leave early and asked if I could come in now. I immediately jumped to it and power-walked over to Biohub! I met briefly with Steve to talk about my work in single-cell alternative splicing and open science before he had to leave. I met then met with some group leaders, a few technicians, and Jim. I think Biohub was at about 20 people at the time so it was pretty empty (we're now at ~80). I had prepared a talk as I had given at other places but didn't end up giving it, only referencing its slides. One of the group leaders who I now work closely with, asked me to define what a *"cell type"* was and I answered that it basically wasn't defineable mathematically with the tools we have now, since we can't say that in Euclidean space of log-transformed counts, if two cells are a distance X or further apart, they are a different cell type. I think I focused on defining cell type as part of a hierarchical tree of the whole dataset.

I was pretty nervous the whole time and was really excited to hear that I received an offer! Coming into this interview, I already had an offer from Startup C and mentioned it during my interview. I negotiated a higher salary than what Startup C offered which sweetened the deal for me since Biohub was where I wanted to work. I wanted to start right away so I got a start date of May 29th, just a few weeks after my May 12th defense date!

#### Green flags:

- Excitement about open science/open source
- Very early stage, so I could have the opportunity to push the culture to be as open science/open source as I wanted

## Summary of how to job-seq

There was a LOT of "right place, right time" kind of thing happening in my job search. I had built up my network over the years through my genuine interests --  open source/open science conferences, contribution to open source software, and blogging. Through my PhD, I built up the [rare and valuable skill](cal newport link) set of applying machine learning algorithms to biology, by working in an RNA biology lab I built up the domain knowledge for biology, and through my own initiative of taking online machine learning courses and reading papers, I learned the technical skills necessary to apply the right algorithm at the right time. I was lucky that I worked on alternative splicing in single cells because it made me stand out from the single-cell RNA-seq crowd, which I have my advisor to thank for giving me a good niche. I was also lucky to have stayed in touch with friends in undergrad and have people around me who could help me prepare for the interviews, for which I'm very grateful.

### Summary of red/"green" flags

I strongly listened to my "gut feeling" about a place because I knew that with great people, it doesn't matter wath the project or the science is, because you'll trust each other enough to be honest and give candid feedback without insulting the other person. It was really important to me that the people I worked with were great, and then the work will come easily.

#### Red flags - not good things

These were things that stood out to me during the interview.

- Interviewer shows off that they're smarter than you
- No answer to "What's your management style?"
- Team doesn't seem to know each other
- Odd weed-out questions
- Dischord between leadership and employees

#### Green flags - good things!

- "No asshole rule"
- "Appreciation ping pong"
- Women in prominent positions in the organization
- Early-stage so I could influence the direction of the company

## Outro

Thanks for reading this very long post. I hope it is helpful to almost-PhDs and others looking for Scientist-level position jobs in industry, as I couldn't find an honest account of the entire process when I was looking.
