Title: You can code, too!
Date: 2013-11-22 14:48:00
Author: sciencemeetproductivity
Category: text
Tags: python, programming, bioinformatics
Slug: 2013-11-22-you-can-code-too

Recently, I've had several conversations with biologists who know they should probably learn to code, but the whole concept is so daunting and they're too busy with their research anyway. I've tried to convince them to pursue programming, not just because it'll make their research analyses faster and better, but because it's an incredibly powerful skill that anyone can learn.

I see programming and engineering as just other methods of creating things that never were, like writing or art or music or dance. Before [Grace Hopper](http://en.wikipedia.org/wiki/Grace_Hopper), there were no compilers for computers, and everything had to be written in machine- and architecture-specific instructions. Before [Donald Knuth](http://www-cs-faculty.stanford.edu/~knuth/) (also see his [advice for young people video](http://www.youtube.com/watch?v=75Ju0eM5T2c)), typesetting math equations was a huge pain, and now we have [TeX](http://www.tug.org/texlive/), [LaTeX](http://en.wikipedia.org/wiki/LaTeX) and derivatives to painlessly create beautiful equations. Before [Linus Torvalds](http://en.wikipedia.org/wiki/Linus_Torvalds) and [Richard Stallman](http://en.wikipedia.org/wiki/Richard_stallman), creators of [GNU/Linux](https://www.gnu.org/gnu/linux-and-gnu.html) (this is the operating systema upon which Mac OS is built), there was no unifying operating system that could be modified and changed for each user's needs. Before Microsoft and Apple computer, the concept of the personal computer didn't even exist, and it was thought to be just a fad that wouldn't hold.

The reason I got into programming wasn't to make games or to hack into things. I started because I was working in a biology lab on protein binding microarrays, and I ran a graduate student's code which turned this glass slide with polka dots of where proteins bound to double-stranded DNA, into the information of the letters bound by this protein. Then, I took my first programming class in my sophomore year of college, and I only took it because it was required for Biological Engineering. It was in [Python](http://www.python.org/). I thought it was pretty cool, but the moment where I truly became a computational biologist was my second research experience where I coded up [Hidden Markov Models](http://www.nature.com/nbt/journal/v22/n10/full/nbt1004-1315.html) in Python. The lab I was working in was a C-language lab so I decided to learn C and "translate" my program from Python to C. It was awesome. As much as `malloc` and `free` were annoying (though `valgrind` helped!), it was so satisfying to see my program run 100x faster in C, and have the satisfaction that I learned this difficult language. Keep in mind that I had been coding for less than a year when I attempted this project.

When I did my [MS in bioinformatics](http://bme.soe.ucsc.edu/) at [UC-Santa Cruz](http://www.ucsc.edu/), I met Paola. She had studied biochemistry at [UCLA](http://www.ucla.edu/) and came to UCSC for the program's bimolecular side, since it technically was _biomolecular engineering and_ bioinformatics. The first class we took for the program was a bioinformatics algorithms course, and this was the first time she had coded in her life. And she did *awesome.* She fell in love with coding as I had, and left gel electrophoresis behind to take discrete math courses to boost her CS knowledge. She now works as an engineer at [Survey Monkey](https://www.surveymonkey.com/), getting hired after just over a year of coding. 

Another friend of mine, [Neha](http://www.nehalita.com/) studied mechanical engineering in undergrad, went into consulting, but hated it. She applied to [Hacker School](https://www.hackerschool.com/) twice [but didn't get in](http://nerdneha.tumblr.com/post/48064403230/try-try-again-a-lesson-in-persistence). She took [Udacity](https://www.udacity.com/) classes on programming and applied again, this time getting in. Through three months of intensive programming and software development at Hacker School, Neha learned more than many people learn in years. She quickly found a job as a software engineer at [Rent the Runway](http://www.renttherunway.com/rtr_home) and adores it.

I'm incredibly proud of Paola and Neha for reinventing themselves. And the task of learning programming is not impossible, it just takes time and [grit](http://www.huffingtonpost.com/tedtalks/angela-lee-duckworth-tedtalk_b_4277459.html). And you can do it, too.

Next time you sit down to analyze some differential expression or calculate some statistics, force yourself to do it in Python, specifically [pandas](http://pandas.pydata.org). Feel free to tweet at me with questions! My handle is [@olgabot](https://twitter.com/olgabot)

## EDIT: 

I forgot to mention that one of my mentors, [Sean Eddy](http://en.wikipedia.org/wiki/Sean_Eddy), studied molecular biology for his PhD, and only after he earned his PhD did he get into coding. And he's made significant contributions to the field of computational biology through the sequence homology comparison program, [HMMER](http://hmmer.janelia.org/), RNA structure prediction through [Infernal](http://infernal.janelia.org/). If you can do it after a PhD, it's truly never too late!

On the flip side, at both UC-Santa Cruz and UC-San Diego there are lifelong software engineers who have spent decades coding up programs, and have decided that they want to apply their programming skills to something else. So they go do a PhD in bioinformatics! Either way, from biology to coding, or from coding to biology, you end up in bioinformatics! Clearly the best field :)

### Resources for learning to code

* [Codecademy](http://www.codecademy.com/) (like a "code-academy"... get it??): Codecademy is an awesome resource for getting started with a language. I used it to learn Javascript!
* [Rosalind.info](http://rosalind.info/problems/locations/): Named after Rosalind Franklin, whose X-ray crystallography experiments helped Watson and Crick discover that DNA is a double helix, Rosalind.info is an awesome resource for bioinformatics-specific problems. You start with translating DNA to RNA to Protein, then eventually you build your own genome assembler!
* [try.github.io](http://try.github.io/levels/1/challenges/1): Cool way to start learning code versioning (keeping backups of your code), which is ultra-important so you don't have to try to remember what you did six months ago.
* [Coursera](https://www.coursera.org/): An online learning platform with tons of classes, including the awesome (Bioinformatics Algorithms)[https://www.coursera.org/course/bioinformatics] class that I helped design. I've also taken the [Probabalistic Graphical Models](https://www.coursera.org/course/pgm) and [Machine Learning](https://www.coursera.org/course/ml) courses. PGMs was good, but ML was a little watered down for my math major self, but it's a good intro overall.
* [Udacity](https://www.udacity.com/): Another online learning platform. They have tons of courses on [Computer Science](https://www.udacity.com/course/cs101), [Web Development](https://www.udacity.com/course/cs253), [Design of Computer Programs](https://www.udacity.com/course/cs212) (taught by [Peter Norvig](http://en.wikipedia.org/wiki/Peter_Norvig)!!!), and many, many, more.
